### PR TITLE
Mailroom OO assignment

### PR DESCRIPTION
**Pull request by archwu**
Hi Ron,

Here is my work done on this final assignment, I am still a little confused that how much my work is object-oriented and the meaning of using a module on this project. I was doing unit tests when all the code was in the same file mailroom_oo.py and they were testing well. However, they are not working after I split them. Let me know if there is a good way to unit test the whole module rather than a single file.

Thanks,
Arch

### CODE REVIEW

File: students/ArchWu/Session8/mailroom_oo.py [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+
+def init_donor_db():
+    db = DonorCollection()
+    db.add_donor(Donor("William Gates III", [653772.32, 12.17]))
+    db.add_donor(Donor("Jeff Bezos", [877.33]))
+    db.add_donor(Donor("Paul Allen", [663.23, 43.87, 1.32]))
+    db.add_donor(Donor("Mark Zuckerberg", [1663.23, 4300.87, 10432.0]))
+    return db
+
+class Donor:
```
Comment (nairraghav) [inline students/ArchWu/Session8/mailroom_oo.py:13]: You already have this called in your donor models, you should import it instead.

File: students/ArchWu/Session8/mailroom_oo.py [hunk 2]
```diff
@@ -0,0 +1,192 @@
+#!/usr/bin/env python3
+import sys, os, math
+from textwrap import dedent
+
```
Comment (nairraghav) [inline students/ArchWu/Session8/mailroom_oo.py:4]: from donor_model import Donor, DonorCollection

File: students/ArchWu/Session8/mailroom_oo.py [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+    for donor in db.donors.items():
+        with open('{}/{}.txt'.format(temp_path, donor[1].name), 'w') as f:
+            num_donation = donor[1].total
+            greetings = 'Dear {},\n'.format(donor[1].name)
+            body = '\nThank you for your generous gift ${} to us!\n'.format(num_donation)
+            ending = '\nSincerely,\n  ABC foundations'
+            f.write(greetings + body + ending)
+    return
+
+def normalize_name(name):
```
Comment (nairraghav) [inline students/ArchWu/Session8/mailroom_oo.py:186]: You already have this in yourDonor object, you can call it from there (Donor.normalize_name)

File: students/ArchWu/Session8/test_mailroom_oo.py [hunk 4]
```diff
@@ -0,0 +1,111 @@
+#!/usr/bin/env python3
+
+"""
+unit tests for the mailroom program
+"""
+import donor_model
```
Comment (nairraghav) [inline students/ArchWu/Session8/test_mailroom_oo.py:6]: if this is import in your mailroom_oo, you will not need to call it here.

**Comment by nairraghav** [comment]
Hey Arch, 

You should be able to have it functional when you import the donor_model file in your mailroom_oo. When you import that and run your tests, what do you see?

**Comment by archwu** [comment]
> Hey Arch,
> 
> You should be able to have it functional when you import the donor_model file in your mailroom_oo. When you import that and run your tests, what do you see?

It was like no test was run and nothing happens

File: students/ArchWu/Session8/mailroom_oo.py (hunk 1 above)
Comment (archwu) [reply students/ArchWu/Session8/mailroom_oo.py:13]: Actually this mailroom_oo is the file I was working on before I saw the requirement to split into multiple files and it happen to have the same file name as the requirement mentioned. What I did try is simply cutting those class initialization code into donor_model and split into two files.

**Comment by nairraghav** [comment]
Hey Arch, I have copied over your files, removed the classes into their own files called "donor_models" and ran the test. It worked successfully for me
Screen Shot 2019-06-15 at 10 59 17 AM

**Comment by nairraghav** [comment]
I will give you credit for what you have since you pretty much have it working ✅

**Pull request merged**